### DISCUSSION TITLE
DAG of flags

### DISCUSSION THREAD

**Discussion by brncsk** [oclif/core#822] [Ideas]
**Is your feature request related to a problem? Please describe.**

I have a use case where I'd like to specify a custom flag's `default` or `options` based on the value passed to other flag(s), or at least have the ability to provide `options` asynchronously.

**Describe the solution you'd like**

Ideally, this would be an extension of `dependsOn`/`relationships` et al., by topologically sorting flags based on their interdependencies. My goal is to be able to use my `inquirer`-based prompter to ask the user for flags that were unspecified on the command line, in the order specified by the dependency graph.

I'm aware this might be out of scope for `oclif`, but I'm willing to provide more context if needed.

**Comment by mdonnalley** [comment]
@brncsk Thanks for the feature request. This is something that we were once interested in for the Salesforce CLI but it proved exceedingly difficult to pull off unfortunately.  I think this is out of scope for oclif at the moment, but I'm going to convert this to a discussion in case anyone else would like upvote this as a feature
